src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.SkeletonReferenceCache.cs

```diff
@@ -218,74 +218,79 @@ private struct SkeletonReferenceCache
         private static SkeletonReferenceSet? CreateSkeletonSet(
             SolutionServices services, Compilation compilation, CancellationToken cancellationToken)
         {
-            var storage = TryCreateMetadataStorage(services, compilation, cancellationToken);
-            if (storage == null)
+            var (metadata, storageHandle) = TryCreateMetadataAndHandle();
+            if (metadata == null)
                 return null;
 
-            var metadata = AssemblyMetadata.CreateFromStream(storage.ReadStream(cancellationToken), leaveOpen: false);
-
             // read in the stream and pass ownership of it to the metadata object.  When it is disposed it will dispose
             // the stream as well.
             return new SkeletonReferenceSet(
                 metadata,
+                storageHandle,
                 compilation.AssemblyName,
                 new DeferredDocumentationProvider(compilation));
-        }
-
-        private static ITemporaryStreamStorageInternal? TryCreateMetadataStorage(SolutionServices services, Compilation compilation, CancellationToken cancellationToken)
-        {
-            cancellationToken.ThrowIfCancellationRequested();
-
-            var logger = services.GetService<IWorkspaceTestLogger>();
 
-            try
+            (AssemblyMetadata? metadata, ITemporaryStorageHandle storageHandle) TryCreateMetadataAndHandle()
```

Comment:
>TryCreateMetadataAndHandle [](http://example.com/codeflow?start=80&length=26)

nit: maybe return bool instead and out the results?

Reply:
i like this.